recognition/advanced_tree_search.py

```diff
@@ -77,20 +77,22 @@ def cleanup_before_run(self, cmd, retry, *args):
     @classmethod
     def find_arpa_lms(cls, lm_config, lm_post_config=None):
         result = []
+
+        def has_image(c, pc):
+            res = c._get("image") is not None
+            res = res or (pc is not None and pc._get("image") is not None)
+            return res
+
         if lm_config.type == "ARPA":
-            result.append((lm_config, lm_post_config))
+            if not has_image(lm_config, lm_post_config):
+                result.append((lm_config, lm_post_config))
         elif lm_config.type == "combine":
             for i in range(1, lm_config.num_lms + 1):
                 sub_lm_config = lm_config["lm-%d" % i]
-                if sub_lm_config.type == "ARPA":
-                    result.append(
-                        (
-                            sub_lm_config,
-                            lm_post_config["lm-%d" % i]
-                            if lm_post_config is not None
-                            else None,
-                        )
-                    )
+                sub_lm_post_config = (
+                    lm_post_config["lm-%d" % i] if lm_post_config is not None else None
+                )
+                result.append(find_arpa_lms(sub_lm_config, sub_lm_post_config))
```

Comment:
Did you test this?^^ 
I think it should be 
```suggestion
                result.append(cls.find_arpa_lms(sub_lm_config, sub_lm_post_config))
```

No wait. `find_arpa_lms` returns already a list (empty if no LMs are found), so we should not append but concatenate
```suggestion
                result += cls.find_arpa_lms(sub_lm_config, sub_lm_post_config)
```

(sorry, my error previously.)

Reply:
I have not. Don't have a setup that uses the combine-lm case^^